nav2_system_tests/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 cmake_minimum_required(VERSION 3.5)
 project(nav2_system_tests)
 
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 23)
+endif()
```

Comment:
added because of:

```
opt/overlay_ws/src/navigation2/nav2_system_tests/src/behaviors/spin/test_spin_behavior_node.cpp:74:3: error: ‘f32’ or ‘F32’ suffix on floating constant only available with ‘-std=c++2b’ or ‘-std=gnu++2b’ [-Werror=pedantic]
   74 |   if (std::getenv("MAKE_FAKE_COSTMAP") != NULL && abs(target_yaw) > M_PI_2f32) {
```